src/modules/iam/nm_iam.h

```diff
@@ -218,6 +244,8 @@ void nm_iam_set_local_initial_pairing(struct nm_iam* iam, bool enabled);
  * @param username [in] the username of the new user
  * @return NM_IAM_ERROR_USER_EXISTS if the specified user already exists.
            NM_IAM_ERROR_OK if the user was created successfully.
+           NM_IAM_ERROR_INVALID_ARGUMENT if the username was too long.
+           NM_IAM_ERROR_INTERNAL if allocation failed or username was invalid
```

Comment:
If allocations fails we need to return 500 "out of resources". If the username is invalid it is a 400 bad request.
Username being too long or invalid are both a form for bad request.